systemd/systemd.go

```diff
@@ -1457,23 +1484,24 @@ func (s *systemd) AddMountUnitFileWithOptions(unitOptions *MountUnitOptions) (st
 	daemonReloadLock.Lock()
 	defer daemonReloadLock.Unlock()
 
-	mountUnitName, err := writeMountUnitFile(unitOptions)
+	mountUnitName, modified, err := writeMountUnitFile(unitOptions)
 	if err != nil {
 		return "", err
 	}
+	if modified != MountClean {
+		// we need to do a daemon-reload here to ensure that systemd really
+		// knows about this new mount unit file
+		if err := s.daemonReloadNoLock(); err != nil {
+			return "", err
+		}
 
-	// we need to do a daemon-reload here to ensure that systemd really
-	// knows about this new mount unit file
-	if err := s.daemonReloadNoLock(); err != nil {
-		return "", err
-	}
-
-	units := []string{mountUnitName}
-	if err := s.EnableNoReload(units); err != nil {
-		return "", err
-	}
-	if err := s.Start(units); err != nil {
-		return "", err
+		units := []string{mountUnitName}
+		if err := s.EnableNoReload(units); err != nil {
+			return "", err
+		}
+		if err := s.ReloadOrRestart(units); err != nil {
```

Comment:
I think that this change (from `Start` to `ReloadOrRestart` needs to be explained better, either as a code comment or in the commit message if it's long).

Reply:
I added a comment.